src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -231,42 +319,53 @@ export default class PhotoField extends React.Component {
     this.setState({ dragAndDropSupported: dragAndDropSupported && !iOS });
   }
 
-  detectWidth = () => {
-    const windowWidth = window.innerWidth;
-    this.setState({ windowWidth });
-  }
-
-  moveUp = () => {
-    this.refs.cropper.move(0, -5);
-  }
+  updateBoundaryWarningAndButtonStates = () => {
+    const moveBoundariesMet = getMoveBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
 
-  moveDown = () => {
-    this.refs.cropper.move(0, 5);
+    this.setState({ ...makeMoveButtonsEnabledStates(moveBoundariesMet) });
+    this.setState({
+      warningMessage: makeCropBoundaryWarningMessage(moveBoundariesMet),
+      ...makeMoveButtonsEnabledStates(moveBoundariesMet)
+    });
   }
 
-  moveRight = () => {
-    this.refs.cropper.move(-5, 0);
+  cropend = () => { // casing matches Cropper argument
+    this.updateBoundaryWarningAndButtonStates();
   }
 
-  moveLeft = () => {
-    this.refs.cropper.move(5, 0);
-  }
-
-  zoom = (e) => {
-    if (e.target.value < MAX_RATIO && e.target.value > MIN_RATIO) {
-      this.refs.cropper.zoomTo(e.target.value);
+  handleMove = (direction) => {
+    switch (direction) {
+      case 'up':
+        this.refs.cropper.move(0, -5);
+        break;
+      case 'down':
+        this.refs.cropper.move(0, 5);
+        break;
+      case 'right':
+        this.refs.cropper.move(5, 0);
+        break;
+      case 'left':
+        this.refs.cropper.move(-5, 0);
+        break;
+      default:
     }
-  }
 
-  zoomIn = () => {
-    if (this.state.zoomValue < MAX_RATIO) {
-      this.refs.cropper.zoom(0.1);
-    }
+    this.updateBoundaryWarningAndButtonStates();
   }
 
-  zoomOut = () => {
-    if (this.state.zoomValue > MIN_RATIO) {
-      this.refs.cropper.zoom(-0.1);
+  handleZoom = (direction) => {
```

Comment:
We're not very consistent about this, but to me the `handleZoom` name implies that we're handling a zoom event, when we're actually triggering a zoom event (we're handling a click on the zoom button(s)). Zoom events are handled in `onZoom`.

I don't have a strong opinion about this either way, but struck me when I was reading the code for handleZoom and handleMove.

Reply:
Good catch- I was working on making these names consistent and forgot to update this one.